### PR TITLE
always show tooltips in preferences

### PR DESCRIPTION
**Pull request by ronso0**
https://bugs.launchpad.net/mixxx/+bug/1840493

### CODE REVIEW

File: src/mixxxmainwindow.cpp [hunk 1]
```diff
@@ -1073,8 +1073,11 @@ bool MixxxMainWindow::loadConfiguredSkin() {
 
 bool MixxxMainWindow::eventFilter(QObject* obj, QEvent* event) {
     if (event->type() == QEvent::ToolTip) {
-        // return true for no tool tips
-        switch (m_toolTipsCfg) {
+        // always show tooltips in the preferences window
+        QWidget* actWin = QApplication::activeWindow();
+        if (actWin && QLatin1String(actWin->metaObject()->className()) != "DlgPreferences") {
```
Comment (ronso0) [inline src/mixxxmainwindow.cpp:1078]: that could also be
``` c++
if (actWin && actWin->objectName() != "DlgPreferencesDlg")
```
both work, I don't know what's preferred. Also, I didn't manage to cast char to QString, the attempts were rather elaborate for a string comparison.

**Comment by Be-ing** [comment]
Rebase on 2.3?

**Comment by ronso0** [comment]
> Rebase on 2.3?

why not. any comment on the code?

File: src/mixxx.cpp [hunk 2]
```diff
@@ -1606,8 +1606,11 @@ void MixxxMainWindow::rebootMixxxView() {
 
 bool MixxxMainWindow::eventFilter(QObject* obj, QEvent* event) {
     if (event->type() == QEvent::ToolTip) {
-        // return true for no tool tips
-        switch (m_toolTipsCfg) {
+        // always show tooltips in the preferences window
+        QWidget* actWin = QApplication::activeWindow();
```
Comment (Be-ing) [inline src/mixxx.cpp:1610]:
```suggestion
        QWidget* activeWindow = QApplication::activeWindow();
```
Please spell out variable names.

**Pull request merged**